kqueen/blueprints/api/views.py

```diff
@@ -31,86 +55,307 @@ def index():
+    obj = get_object(Provisioner, pk)
 
-    # load object
     try:
-        obj = Cluster.load(object_id)
-    except NameError:
-        abort(404)
+        obj.delete()
+    except:
+        abort(500)
 
-    return jsonify(obj.status())
+    return jsonify({'id': obj.id, 'state': 'deleted'})
+
+
+# Organizations
+
+@api.route('/organizations', methods=['GET'])
+@jwt_required()
+def organization_list():
+    output = []
+
+    for obj in list(Organization.list(return_objects=True).values()):
+        output.append(obj.get_dict())
+
+    return jsonify(output)
 
 
-@api.route('/clusters/<cluster_id>/topology-data', methods=['GET'])
-def cluster_topology_data(cluster_id):
+@api.route('/organizations', methods=['POST'])
+@jwt_required()
+def organization_create():
+    if not request.json:
+        abort(400)
+    else:
+        obj = Organization(**request.json)
+        try:
+            obj.save()
+            output = obj.serialize()
+        except:
+            abort(500)
+
+    return jsonify(output)
+
+
+@api.route('/organizations/<pk>', methods=['GET'])
+@jwt_required()
+def organization_get(pk):
+    obj = get_object(Organization, pk)
+
+    return jsonify(obj.get_dict())
+
+
+@api.route('/organizations/<pk>', methods=['PATCH'])
+@jwt_required()
+def organization_update(pk):
+    if not request.json:
+        abort(400)
+
+    data = request.json
+    if not isinstance(data, dict):
+        abort(400)
+
+    obj = get_object(Organization, pk)
+    for key, value in data.items():
+        setattr(obj, key, value)
 
     try:
-        object_id = UUID(cluster_id, version=4)
-    except ValueError:
-        abort(404)
+        obj.save()
+        return jsonify(obj.serialize())
+    except:
+        abort(500)
+
+
+@api.route('/organizations/<pk>', methods=['DELETE'])
+@jwt_required()
+def organization_delete(pk):
+    obj = get_object(Organization, pk)
 
-    # load object
     try:
-        obj = Cluster.load(object_id)
-    except NameError:
-        abort(404)
+        obj.delete()
+    except:
+        abort(500)
 
-    return jsonify(obj.topology_data())
+    return jsonify({'id': obj.id, 'state': 'deleted'})
+
+
+# Users
+
+@api.route('/users', methods=['GET'])
+@jwt_required()
+def user_list():
+    output = []
+
+    for obj in list(User.list(return_objects=True).values()):
+        output.append(obj.get_dict())
+
+    return jsonify(output)
 
 
-@api.route('/clusters/<cluster_id>/kubeconfig', methods=['GET'])
+@api.route('/users', methods=['POST'])
 @jwt_required()
-def cluster_kubeconfig(cluster_id):
+def user_create():
+    if not request.json:
+        abort(400)
+    else:
+        obj = User(**request.json)
+        try:
+            obj.save()
+            output = obj.serialize()
+        except:
+            abort(500)
+
+    return jsonify(output)
+
+
+@api.route('/users/<pk>', methods=['GET'])
+@jwt_required()
+def user_get(pk):
+    obj = get_object(User, pk)
+
+    return jsonify(obj.get_dict())
```

Comment:
It would be nice if this could return related object organization as nested dict